salon_creator/lib/screens/lesson_list_screen.dart

```diff
@@ -1,15 +1,54 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:flutter/material.dart';
+import 'package:salon_creator/firebase/database.dart';
+import 'package:salon_creator/models/creator_model.dart';
+import 'package:salon_creator/widgets/lesson_card.dart';
 
 class LessonListScreen extends StatefulWidget {
-  const LessonListScreen({Key key}) : super(key: key);
+  const LessonListScreen({Key key, this.creator}) : super(key: key);
+  final CreatorModel creator;
 
   @override
   _LessonListScreenState createState() => _LessonListScreenState();
 }
 
 class _LessonListScreenState extends State<LessonListScreen> {
+  List<Widget> lessons = [];
+  final dbHanlder = DbHandler();
+
   @override
   Widget build(BuildContext context) {
-    return Container();
+    return StreamBuilder<QuerySnapshot>(
+      stream: FirebaseFirestore.instance
+          .collection(DbHandler.salonsCollection)
+          .doc(widget.creator.salonId)
+          .collection(DbHandler.lessonsCollection)
+          .snapshots(),
+      builder: (BuildContext context, AsyncSnapshot<QuerySnapshot> snapshot) {
+        if (!snapshot.hasData) {
+          return LinearProgressIndicator();
+        }
+        return Padding(
+          padding: EdgeInsets.only(top: 12.0),
+          child: ListView(
+            children: snapshot.data.docs.map((DocumentSnapshot document) {
+              Map<String, dynamic> data =
+                  document.data() as Map<String, dynamic>;
+              return Padding(
+                padding: EdgeInsets.symmetric(horizontal: 10.0),
+                child: LessonCard(
+                  isPublished: data['isPublish'],
```

Comment:
this translation from data to class should be done inside the LessonCard constructor, like we do in other places.